libs/cardano-ledger-core/src/Cardano/Ledger/Core/Era.hs

```diff
@@ -130,6 +132,8 @@ type family EraRule (rule :: Symbol) era = (r :: Type) | r -> rule
 -- does not provide for us unfortunately.
 type family EraRuleFailure (rule :: Symbol) era = (r :: Type) | r -> rule era
 
+type family EraRuleEvent (rule :: Symbol) era = (r :: Type) | r -> rule era
```

Comment:
If it's not too difficult, I think it would be nice to have this change and the changes that it entails in a separate commit, because there seems to be a lot happening here.

Reply:
I think it'd be too much work to split this now, but I'll try to keep that in mind next time